packages/core/core/src/Parcel.js

```diff
@@ -92,6 +95,41 @@ export default class Parcel {
 
     this.#runPackage = this.#farm.mkhandle('runPackage');
 
+    await this.initializeWatcher();
+  }
+
+  async initializeWatcher() {
+    let projectRoot = this.#resolvedOptions.projectRoot;
+    if (this.#resolvedOptions.watch) {
+      // TODO: ideally these should all be absolute paths already set up on #resolvedOptions
+      let targetDirs = this.#resolvedOptions.targets.map(target =>
+        path.resolve(process.cwd(), target.distDir)
+      );
+      let cacheDir = path.resolve(
+        process.cwd(),
+        this.#resolvedOptions.cacheDir
+      );
+      let vcsDirs = ['.git', '.hg'].map(dir => path.join(projectRoot, dir));
```

Comment:
This would not handle the case where the vcs folder is nested deeply inside of the project root, but that does not seem common. It could be outside of the project root, but we won't be watching outside of project root, so it shouldn't matter.